pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTenantRestletResource.java

```diff
@@ -103,6 +112,9 @@ public class PinotTenantRestletResource {
   @Inject
   ControllerMetrics _controllerMetrics;
 
+  @Inject
+  ExecutorService _executorService;
```

Comment:
I'm not sure if using the same shared executorService (which is also being used to handle multi get API call responses) for parallelly running multiple blocking operation like table rebalance is a good idea.. Although I do see same executorService being used to run rebalance for single table.
Thoughts?

Reply:
Makes sense. Even I referred to the single table rebalance implementation but agree with you point that it will affect the shared executorService i cases where the tenant has lot of tables and user runs the rebalance with high parallelism.